### PR TITLE
macOS: add basic menubar support

### PR DESCRIPTION
**Pull request by selsta**
menubar

### CODE REVIEW

**Comment by jonathancross** [comment]
@selsta Needs a rebase.

**Comment by selsta** [comment]
Rebased, but there’s still a bug that breaks Windows and Linux compilation :/

**Comment by selsta** [comment]
Due to a bug in Qt 5.9, this is broken on Windows and Linux. PR will have to stay open until we update to a newer Qt version.

**Comment by selsta** [comment]
This passes CI now, hmm... I wonder if we could merge this now.

**Comment by selsta** [comment]
Still have to work on adding cmake support to this PR.

**Comment by selsta** [comment]
Have to add static (linux) support

https://invent.kde.org/network/kaidan/-/blob/f597b3f2d587aee8fc93ee2a77a4f7ed08a2dc4c/CMakeLists.txt

**Comment by selsta** [comment]
Fixed all issues now, ready for review.

File: components/MenuBar.qml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+// MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+// THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+// STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+// THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import Qt.labs.platform 1.0 as PlatformLabs
+import "../components" as MoneroComponents
```
Comment (xiphon) [inline components/MenuBar.qml:30]: ?

Comment (selsta) [reply components/MenuBar.qml]: updated (if that's what you meant)

**Review by xiphon — APPROVED** [review]

**Pull request merged**